### PR TITLE
Add --stdout option / redirect error messages to stderr

### PR DESCRIPTION
**Pull request by sairion**
Hi,

I added an option to take output to stdout. This is my first attempt to write Rust code other than hello world, so I believe it could be not-so-much-idiomatic ;) Please take your time to tell me if there's anything to fix!

Also, I redirected error messages to stderr to prevent error messages to be written to stdout. I needed this option because I wanted to replace svgo with svgcleaner, using stream interface in Node.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -59,11 +60,11 @@ fn main() {
     let write_opt = cli::gen_write_options(&args);
     let cleaning_opt = cli::gen_cleaning_options(&args);
 
-    let in_file  = args.value_of("in-file").unwrap();
-    let out_file = args.value_of("out-file").unwrap();
+    let in_file = args.value_of("in-file").unwrap();
+    let out_file = args.value_of("out-file").unwrap_or("");
```
Comment (RazrFalcon) [inline src/main.rs]: I think we should use `Option<String>` in this case. So `.unwrap_or("")` is not needed.

File: src/main.rs [hunk 2]
```diff
@@ -129,21 +130,25 @@ fn main() {
...
 
     // save buffer
-    try_msg!(cleaner::save_file(&buf[..], out_file));
+    if cli::get_flag(&args, Key::Stdout) {
+        try_msg!(cleaner::write_stdout(&buf[..]));
+    }
 
-    if cli::get_flag(&args, Key::Quiet) {
-        return;
+    if !out_file.is_empty() {
```
Comment (RazrFalcon) [inline src/main.rs]: Since we switched to `Option<String>` it will became:
`if let Some(out_file) = out_file {`

**Comment by RazrFalcon** [comment]
Looks good, but there are few problems:

1. Warnings from `svgcleaner`, `svgdom` and `svgparser` will still be printed to `stdout`. So we have to update all of them. It's not a problem, just a task. I planned to do this one day. So I should update all the dependencies and only then merge this patch.
1. stdout usually marked as `-`, `--`, `-c` or `--stdout`. I don't know if `clap` supports `-`/`--`, but `-c` should be added.
1. I use custom help text, so you should update `data/help.txt` as well.

As for code itself - I've added few comments.

File: src/main.rs [hunk 3]
```diff
@@ -72,7 +73,7 @@ fn main() {
 
     let on_err = || {
         // copy original file to destination
-        if cli::get_flag(&args, Key::CopyOnError) {
+        if out_file.is_empty() && cli::get_flag(&args, Key::CopyOnError) {
```
Comment (RazrFalcon) [inline src/main.rs]: Is there an error? It should be `!out_file.is_empty()`?

```diff
-out_file.is_empty()
+out_file.is_some()
```

**Comment by sairion** [comment]
@RazrFalcon I believe `--` option is a bit different with standard output flag? It seems clap (kinda) [supports it](https://docs.rs/clap/2.22.2/clap/struct.Arg.html#last.v) but I don't think it is applicable here.

I've add `-c` as positional argument and set `--stdout` as alias of it. Please take a look! 😄

File: src/cleaner.rs [hunk 4]
```diff
@@ -207,6 +207,11 @@ pub fn write_buffer(doc: &Document, opt: &WriteOptions, buf: &mut Vec<u8>) {
     doc.write_buf_opt(opt, buf);
 }
 
+pub fn write_stdout(data: &[u8]) -> Result<(), io::Error> {
+    io::stdout().write(&data).unwrap();
+    Ok(())
```
Comment (RazrFalcon) [inline src/cleaner.rs]: You should remove either `Ok(())` or `unwrap()`, because this code has no meaning.

File: src/main.rs [hunk 5]
```diff
@@ -72,10 +74,10 @@ fn main() {
 
     let on_err = || {
         // copy original file to destination
-        if cli::get_flag(&args, Key::CopyOnError) {
+        if out_file.is_some() && cli::get_flag(&args, Key::CopyOnError) {
             // copy a file only when paths are different
-            if in_file != out_file {
-                try_msg!(fs::copy(in_file, out_file));
+            if in_file != out_file.unwrap() {
+                try_msg!(fs::copy(in_file, out_file.unwrap()));
```
Comment (RazrFalcon) [inline src/main.rs]: Too much `unwrap`. You can you `if let` syntax here too.

```rust
if let Some(out_file) = out_file {
    if in_file != out_file {
        try_msg!(fs::copy(in_file, out_file));
    }
}
```

**Comment by RazrFalcon** [comment]
I've added two comments. You can fix this issues and squash commits into one. Then I will merge it.

**Comment by RazrFalcon** [comment]
`svgdom` and `svgparser` will print warnings to `stderr` now.

**Comment by sairion** [comment]
@RazrFalcon fixed, and squashed commits into one (I didn't see your commit mixed in, sorry)

**Comment by RazrFalcon** [comment]
Good. Thanks.

**Pull request merged**